umbraco/Umbraco.AuthorizedServices
# Refresh token with client credentials flow and use async for data access.

This PR was initiated with the request in issue https://github.com/umbraco/Umbraco.AuthorizedServices/issues/42, where it looks like we missed a step with refreshing the token with the client credentials flow.

Hence the main part that needs review is the `EnsureAccessToken` method.

Then I also realised we weren't making use of async in the data access calls, so I added that too.  Hence most of the PR is changes related to that.

To Test:

- A general check to make sure existing calls to authenticate and use an authorized service still work.
- Specifically a check on the client credentials flow.  I don't have an example account, but I'm assuming you do @acoumb, so I'd suggest testing with:
    - Authorize and use a service using the OAuth2 client credentials flow.
    - Update the token record in the database to set it to have expired.
    - Check that a new token is seamlessly requested when you make another call to the service.
